chainer/computational_graph.py

```diff
@@ -11,28 +11,24 @@ class DotNode(object):
     with some utilities for dot language.
 
     """
-    def __init__(self, node):
+    def __init__(self, node, attribute=None):
         """Initializes DotNode.
 
         Args:
             node: :class: `Variable` object or :class: `Function` object.
+            attribute (dict): Attributes for the node.
 
         """
         assert isinstance(node, (variable.Variable, function.Function))
         self.node = node
         self.id_ = id(node)
-        self.attribute = {
-            "label": self.node.label,
-            "shape": self._shape()
-        }
-
-    def _shape(self):
-        """Returns shape type of node."""
-
-        if isinstance(self.node, variable.Variable):
-            return "oval"
+        if isinstance(node, variable.Variable):
+            self.attribute = {'shape': 'oval'}
         else:
-            return "box"
+            self.attribute = {'shape': 'box'}
+        if self.attribute is not None:
+            self.attribute.update(attribute)
+        self.attribute['label'] = node.label
```

Comment:
I intended to set  `label` value in `attribute` as a fixed value (`node.label`).
But as you said, it can be updated with attribute, I will change it.